WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/Bulk Edit Price/BulkUpdatePriceViewController.swift

```diff
@@ -0,0 +1,193 @@
+        title = viewModel.screenTitle()
+        view.backgroundColor = .listBackground
+    }
+
+    /// Setup receiving updates for data changes from the view model
+    ///
+    private func configureViewModel() {
+        viewModel.$saveButtonState.sink { [weak self] state in
+            self?.saveButton.isEnabled = state == .enabled
+
+            if state == .loading {
+                self?.saveButton.showActivityIndicator()
+                // Dismiss the keyboard while we are in loading state
+                self?.view.endEditing(true)
+            } else {
+                self?.saveButton.hideActivityIndicator()
+            }
+        }.store(in: &subscriptions)
+    }
+
+    /// Configures the table view: registers Nibs & setup dataSource
+    ///
+    private func configureTableView() {
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.backgroundColor = .listBackground
+
+        registerTableViewCells()
+
+        tableView.dataSource = self
+    }
+
+    private func registerTableViewCells() {
+        for row in Row.allCases {
+            tableView.registerNib(for: row.type)
+        }
+    }
+
+    /// Called when the save button is tapped to update the price for all variations
+    ///
+    @objc private func saveButtonTapped() {
+        viewModel.saveButtonTapped()
+    }
+}
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension BulkUpdatePriceViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return viewModel.sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.sections[section].rows.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let row = viewModel.sections[indexPath.section].rows[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: row.reuseIdentifier, for: indexPath)
+        configure(cell, at: indexPath)
+
+        return cell
+    }
+
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        return viewModel.sections[section].footer
+    }
+}
+
+// MARK: - Cell configuration
+//
+private extension BulkUpdatePriceViewController {
+    /// Configures a cell
+    ///
+    func configure(_ cell: UITableViewCell, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as UnitInputTableViewCell:
+            let cellViewModel = Product.createRegularPriceViewModel(regularPrice: viewModel.currentPrice,
+                                                                    using: ServiceLocator.currencySettings) { [weak self] value in
+                self?.viewModel.handlePriceChange(value)
+            }
```

Comment:
I know this is not your code but it does not feel right for a `Product` to create a view model. 😅

Reply:
Yes I know, it felt out of place, should I open an improvement issue for this? I think an extension on the view model should be better.

I have a followup PR that adds the proper style (to match the mockups with the input on the left) so I can improve this by moving it to an extension in the view model.